### PR TITLE
fixes for test flakyness post 0.0.4 release

### PR DESCRIPTION
**Pull request by surajkota**
Description of changes:
- there is a slight delay sometimes when tags are available on resource post creation
- adopted_endpoint resource was missing deleting the cr which caused false positive error message clutter in logs
- Job stopping sometimes takes a longer time than timeout
- minor linting and clean up

Testing:
PR build

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/__init__.py [hunk 1]
```diff
@@ -313,3 +313,35 @@ def assert_training_status_in_sync(training_job_name, reference, expected_status
...
+
+
+def get_sagemaker_endpoint_config(config_name: str):
+    try:
+        return sagemaker_client().describe_endpoint_config(
+            EndpointConfigName=config_name
+        )
+    except botocore.exceptions.ClientError as error:
+        logging.error(
+            f"SageMaker could not find a config with the name {config_name}. Error {error}"
```
Comment (ryansteakley) [inline test/e2e/__init__.py:345]: Maybe specify its an endpoint config

**Review by ryansteakley — APPROVED** [review]

**Review by akartsky — APPROVED** [review]

**Pull request merged**